### DISCUSSION TITLE
Why is `baseURL` unset in `getResourceByUrl`?

### DISCUSSION THREAD

**Discussion by nitwhiz** [Gabb-c/pokenode-ts#887] [General]
`getResourceByUrl` requests without the base-url set in the client used for this request. Why is it designed this way?

https://github.com/Gabb-c/pokenode-ts/blob/6572881e692bd6e305e8c8f37851fb892a171a68/src/clients/utility.client.ts#L50-L54

This makes it hard to use this client with a selfhosted instance of the PokeAPI JSON-files, where following the urls in the responses is necessary.

**Comment by Gabb-c** [comment]
Hey @nitwhiz, I appreciate your input and apologize for the delayed response.

Initially, the decision to remove the `baseURL` from this method was made with the intention of facilitating the retrieval of resources from sources other than the default one.

It may be advantageous to consider the inclusion of an optional parameter that allows for the overriding of the `baseURL`, if necessary.

**Reply by nitwhiz** [reply]
an optional parameter sounds good.

**Comment by Gabb-c** [comment]
Sorry for the silence. I was away from the project for a while for personal
reasons, but I'm back and maintaining it actively again. 2.0.0 shipped with native
`fetch`, no runtime dependencies, a new cache and logger API, updated typings, and
weekly drift tests against the live [PokéAPI](https://github.com/pokeapi/pokeapi).

Wrote up the details [here](https://github.com/Gabb-c/pokenode-ts/discussions/1399).

Closing this one — if you've got something specific you're stuck on, open an issue
or reply on that thread and I'll pick it up.
